src/app.ts

```diff
@@ -13,14 +13,37 @@ interface TemperatureSummary {
   average: number
 }
 
+let readingsGlobal: TemperatureReading[]
 export function processReadings(readings: TemperatureReading[]): void {
   // add here your code
-}
 
+  readingsGlobal = readings
+  return
+}
 export function getTemperatureSummary(
   date: Date,
   city: string,
 ): TemperatureSummary | null {
   //add here your code
-  return null
+  const filter = readingsGlobal.filter(
+    (data) => data.city === city && data.time.getTime() === date.getTime(),
+  )
+  if (filter.length < 1) {
+    return null
+  }
+  const TemperatureSummary = {
+    first: filter[0].temperature,
+    last: filter.slice(-1)[0].temperature,
+    high: filter.reduce(
+      (accu, curr) => Math.max(accu, curr.temperature),
+      -Infinity,
+    ),
+    low: filter.reduce(
+      (accu, curr) => Math.min(accu, curr.temperature),
+      Infinity,
+    ),
+    average:
+      filter.reduce((accu, curr) => accu + curr.temperature, 0) / filter.length,
```

Comment:
what happen when we run the division and filter.length is 0. should you check filter.lenght first?
```suggestion
      filter.reduce((accu, curr) => accu + curr.temperature, 0) / filter.length,
```

Reply:
no, actually it will never be 0, because I do a return if filter.length is 0 before this part of the code.